src/realm/sync/client.cpp

```diff
@@ -1226,17 +1226,27 @@ void SessionWrapper::report_progress()
     ClientHistory::get_upload_download_bytes(m_db.get(), downloaded_bytes, downloadable_bytes, uploaded_bytes,
                                              uploadable_bytes, snapshot_version);
 
+    // In protocol versions 25 and earlier, downloadable_bytes was the total
+    // size of the history. From protocol version 26, downloadable_bytes
+    // represent the non-downloaded bytes on the server. Since the user supplied
+    // progress handler interprets downloadable_bytes as the total size of
+    // downloadable bytes, this number must be calculated.  We could change the
+    // meaning of downloadable_bytes for the progress handler, but that would be
+    // a breaking change. Note that protocol version 25 (and earlier) is no
+    // longer supported by clients.
+    std::uint_fast64_t total_bytes = downloaded_bytes + downloadable_bytes;
```

Comment:
I dislike that we're layering this information so that something that's higher in the call stack of the same statically linked library is unaware of the meaning of the information being passed to it. I think this adds complexity for no reason and makes this confusing and error prone.

Reply:
I don't particularly think that this is the right place to do this, but only because I think it should be done even earlier. The exact details of how progress is represented in the protocol are not relevant to anything which isn't interacting directly with the protocol, which includes most of the sync client.